include/ttmlir/Dialect/TTNN/IR/TTNNOpsBackendInterfaces.td

```diff
@@ -0,0 +1,50 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TTMLIR_TTMLIR_DIALECT_TTNN_TTNNOPSPERFINTERFACES_TD
+#define TTMLIR_TTMLIR_DIALECT_TTNN_TTNNOPSPERFINTERFACES_TD
+
+include "mlir/IR/OpBase.td"
+
+def TTNNOpBackendInterface : OpInterface<"TTNNOpBackend"> {
+    let description = [{
+        Interface to access a registered method to infer the return types for an
+        operation that can be used during type inference.
+    }];
+    let cppNamespace = "::mlir::tt::ttnn";
+    let methods = [
+        InterfaceMethod<
+            /*desc=*/[{
+                Return the op kernel estimate in clock cycles.
+            }],
+            /*retTy=*/"size_t",
+            /*methodName=*/"getOpPerfCycles",
+            /*args=*/(ins "const tt::LayoutAttr&":$output_layout), // Subject to change
+            /*methodBody=*/"",
+            /*defaultImplementation=*/"return std::numeric_limits<size_t>::max();"
+        >,
+        InterfaceMethod<
+            /*desc=*/[{
+                Return the op kernel estimate in clock cycles.
+            }],
+            /*retTy=*/"size_t",
+            /*methodName=*/"getOpL1Usage",
+            /*args=*/(ins "const tt::LayoutAttr&":$output_layout), // Subject to change
+            /*methodBody=*/"",
+            /*defaultImplementation=*/"return 0;"
+        >,
+        InterfaceMethod<
+            /*desc=*/[{
+                Return the op kernel estimate in clock cycles.
+            }],
+            /*retTy=*/"bool",
+            /*methodName=*/"isOpLegal",
+            /*args=*/(ins "const tt::LayoutAttr&":$output_layout), // Subject to change
+            /*methodBody=*/"",
+            /*defaultImplementation=*/"return true;"
+        >,
```

Comment:
If this is going to be calling graph capture, which is costly, we might want to have these combined in a single call? @nobradovictt @mbezuljTT Thoughts?

Reply:
per offline discussion, it would makes sense with current implementation in mind. logically it makes sense to keep them separated. let's keep them as is, and revisit this question later. we might cache graph capture calls anyway.